### PR TITLE
Add io.github.philipk.boilr

### PR DESCRIPTION
**Pull request by PhilipK**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.philipk.boilr.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Philip Kristoffersen 2022 <philipkristoffersen@gmail.com> -->
+<component type="desktop-application">
+  <id>io.github.philipk.boilr</id>
+  <metadata_license>CC0</metadata_license>
```
Comment (hfiguiere) [inline io.github.philipk.boilr.appdata.xml:5]: this is an invalid SPDX identifier. See https://spdx.dev/licenses/

File: io.github.philipk.boilr.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    build-commands:
+      - install -Dm644 io.github.philipk.boilr.desktop -t /app/share/applications/
+
+  - name: boilr-appinfo
+    buildsystem: simple
+    sources:
+      - type: file
+        path: io.github.philipk.boilr.appdata.xml
+    build-commands:
+      - install -Dm644 io.github.philipk.boilr.appdata.xml -t /app/share/metainfo/
```
Comment (hfiguiere) [inline io.github.philipk.boilr.yml:70]:
```suggestion
      - cargo-lock.json
      - type: file
        path: io.github.philipk.boilr.desktop
      - type: file
        path: io.github.philipk.boilr.appdata.xml
    build-commands:
      - cargo --offline fetch --manifest-path Cargo.toml --verbose
      - cargo --offline build --release --verbose
      - install -Dm755 ./target/release/boilr -t /app/bin/
      - install -Dm644 resources/io.github.philipk.boilr.png  -t /app/share/icons/hicolor/32x32/apps
      - install -Dm644 io.github.philipk.boilr.desktop -t /app/share/applications/
      - install -Dm644 io.github.philipk.boilr.appdata.xml -t /app/share/metainfo/      
```


Also 32x32 is too small for icons.


And for a future release the .desktop and .appdata.xml should be managed upstream and installed from there.

File: io.github.philipk.boilr.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  <project_license>MIT/APACHE</project_license>
```
Comment (hfiguiere) [inline io.github.philipk.boilr.appdata.xml:6]: nor is this one.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Doesn't build as is.

File: io.github.philipk.boilr.appdata.xml (hunk 1 above)
Comment (PhilipK) [reply io.github.philipk.boilr.appdata.xml:5]: The project is MIT or Apache, whatever the user wants.
For this I will just pick MIT-0 .
But just out of curiosity, is it possible to select multiple?

Comment (hfiguiere) [reply io.github.philipk.boilr.appdata.xml:5]: `CC0` is not a valid identifier. In the list I linked it is `CC0-1.0`.

use `AND` or `OR` to combine multiple license. Identifiers from the list, again.

File: io.github.philipk.boilr.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  - --env=PATH=/app/bin:/usr/bin
+
+modules:
+  - shared-modules/glu/glu-9.json
+
+  - name: boilr
+    buildsystem: simple
+    sources:
+      - type: dir
+        path: ../
```
[the remark this answers is not in the document]
Comment (PhilipK) [reply io.github.philipk.boilr.yml:40]: Ah, ofc, I think something just clicked for me, now I get how flathub actually will build things.
I will get this changed to a type: git and then update this yml.

Thank you for your patience, I see now that this pull request is pretty broken in its current form.

[the remark this answers is not in the document]
Comment (PhilipK) [reply io.github.philipk.boilr.yml]: This has now been changed to a git source

File: io.github.philipk.boilr.appdata.xml (hunk 1 above)
Comment (PhilipK) [reply io.github.philipk.boilr.appdata.xml]: I have moved this to upstream instead

File: io.github.philipk.boilr.appdata.xml (hunk 3 above)
Comment (PhilipK) [reply io.github.philipk.boilr.appdata.xml]: this has been moved to upsteam

**Comment by hfiguiere** [comment]
bot, build io.github.philipk.boilr

**Comment by PhilipK** [comment]
I have also added more icons

**Comment by hfiguiere** [comment]
the only required size is 128x128 or a scalable SVG. FYI.

**Comment by PhilipK** [comment]
The error that failed the build has been fixed upstream

**Comment by hfiguiere** [comment]
you can also:

bot, build io.github.philipk.boilr

**Comment by PhilipK** [comment]
bot, build io.github.philipk.boilr

**Pull request closed without merging**